src/lib/utils.js

```diff
@@ -46,26 +47,129 @@ function isPermissionReadOnly(perm) {
   )
 }
 
+/**
+ * Finds the permission block for the the file
+ * in the permissions owned by the current cozy-client.
+ *
+ * Iterates through parent folders if needed
+ * until we can find the permissions attached to the share
+ *
+ * @private
+ * @param {string} fileId
+ * @param {string|undefined} parentId
+ */
+async function findPermissionFor(fileId, { parentId, client, permissions }) {
+  const perm = permissions.find(perm => {
+    !perm.values || perm.values.includes(fileId)
+  })
+  if (perm) {
+    return perm
+  } else if (parentId) {
+    const parentData = await getFile(client, parentId)
+    const parentParentId = parentData.attributes.dir_id
+    const context = { parentId: parentParentId, client, permissions }
+    return findPermissionFor(parentId, context)
+  } else {
+    return undefined
```

Comment:
https://github.com/cozy/cozy-guidelines/#return-value

Also I find it more readable that way : 
```
if(perm) return perm
if(!parentId) return null
else ... 
```